dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/ajax/CMSMaintenanceAjax.java

```diff
@@ -429,25 +435,22 @@ public void createXMLFiles() throws ServletException, IOException, DotDataExcept
 			Set<Class> tablesToDump = new HashSet<Class>();
 			try {
 
-				/* get a list of all our tables */
-				final Map map = new HashMap();
-
-				//Including Identifier.class because it is not mapped with Hibernate anymore
-				map.put(Identifier.class, null);
-
-				map.putAll(HibernateUtil.getSession().getSessionFactory().getAllClassMetadata());
-
-				Iterator it = map.entrySet().iterator();
-				while (it.hasNext()) {
-					Map.Entry pairs = (Map.Entry) it.next();
-					Class x = (Class) pairs.getKey();
-					if (!x.equals(Inode.class) && !x.equals(Clickstream.class) && !x
-							.equals(ClickstreamRequest.class)
-							&& !x.equals(Plugin.class) && !x.equals(PluginProperty.class)) {
-						tablesToDump.add(x);
-					}
 
-				}
+			    final Set ignoreClasses =  ImmutableSet.of(Inode.class, Clickstream.class,ClickstreamRequest.class,PluginProperty.class,Structure.class,Field.class,FieldVariable.class,PermissionReference.class);
+			    
+			    
+
+                //Including Identifier.class because it is not mapped with Hibernate anymore
+                tablesToDump.add(Identifier.class);
+                tablesToDump.add(MultiTree.class);
+                tablesToDump.add(Tree.class);
+                
+                
+                HibernateUtil.getSession().getSessionFactory().getAllClassMetadata().keySet().stream().forEach(key -> tablesToDump.add((Class) key));
+                
+                ignoreClasses.forEach(clazz -> tablesToDump.remove(clazz));

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "Set<Class>" cannot contain a "E" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2175)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'clazz' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)